controllers/upgrade_handlers.go

```diff
@@ -301,6 +298,66 @@ func (r *ImageBasedUpgradeReconciler) startBackup(ctx context.Context, backupCRs
 	return doNotRequeue(), nil
 }
 
+// nolint:unparam
+func (r *ImageBasedUpgradeReconciler) postPivot(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	r.Log.Info("Starting health check for different components")
+	err := healthChecks(r)
+	if err != nil {
+		upgradeFailedStatus(ibu, err.Error())
+		return requeueWithError(err)
+	}
+
+	return doNotRequeue(), nil
+}
+
+func healthChecks(r *ImageBasedUpgradeReconciler) error {
```

Comment:
Can this be a common utility function, rather than ImageBasedUpgradeReconciler? I may be able to use this in SeedGen, as well

Reply:
Updated should be easy to use now